assignment+4/Record.java

```diff
@@ -1,60 +1,99 @@
 // a custom class that represents each unique record from the GNIS file
-package assignment4;
-
-public class Record {
-  public String cgndbId;
-  public String geographicName;
-  public String genericTerm;
-  public float latitude;
-  public float longitude;
-  public String location;
-
-  // constructor for Record object
-  public Record(String l) {
-    // regex allows us to split only on commas that are outside quotes
-    String[] elements = l.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)");
-    int size = elements.length;
-
-    this.cgndbId = elements[0];
-    this.geographicName = elements[1];
-    this.genericTerm = elements[2];
-    this.latitude = Float.parseFloat(elements[3]);
-    this.longitude = Float.parseFloat(elements[4]);
-    this.location = null;
-
-    // if there is a location, assign it to the class variable
-    if (size == 6) {
-      this.location = elements[5];
-    }
+
+public class Record implements Comparable<Record> {
```

Comment:
What does Comparable<Record> do?

Reply:
If we want to override and introduce custom behaviour to the Record class when we do something like:

`Record record1 = new Record(someID1);
Record record2 = new Record(someID2);
int = record1.compareTo(record2);`

in java, we have to implement the Comparable Interface. line 3 in plain english is saying -> I want a publically available class called "Record" which implements the Comparable interface of type Record.